gremlin-dotnet/src/pom.xml

```diff
@@ -53,16 +53,25 @@ limitations under the License.
                             <goal>execute</goal>
                         </goals>
                         <configuration>
+                            <properties>
+                                <property>
+                                    <name>projectBaseDir</name>
+                                    <value>${project.basedir}</value>
+                                </property>
+                                <property>
+                                    <name>mavenVersion</name>
+                                    <value>${project.version}</value>
+                                </property>
+                            </properties>
                             <scripts>
                                 <script>
-def mavenVersion = "${project.version}"
-def file = new File("${project.basedir}/Gremlin.Net/Gremlin.Net.csproj")
+def file = new File(projectBaseDir,"Gremlin.Net/Gremlin.Net.csproj")
 file.write(file.getText("UTF-8").replaceFirst(/&lt;Version&gt;(.*)&lt;\/Version&gt;/, "&lt;Version&gt;" + mavenVersion + "&lt;/Version&gt;"))
 
-file = new File("${project.basedir}/Gremlin.Net.Template/Gremlin.Net.Template.csproj")
```

Comment:
Interesting. Could you mention what was happening here? Why does using a `<property>` work but the variable substitution not?

Reply:
The issue arises from when the string interpolation happens. If the "${project.basedir}/..." is evaluated by maven then gradle sees the unescaped string and must parse it.
If the `<property>` approach is used, then the [gmavenplus-plugin|https://groovy.github.io/GMavenPlus/plugin-info.html] creates a gradle string variable.

Reply:
that makes sense. it will definitely be crucial that we get that Windows CI build going or we'll mess that up again.